### PR TITLE
Follow up fixes for authorization filtering

### PR DESCRIPTION
**Pull request by Geal**
Follow up for missing cases in #5952

---

**Checklist**

Complete the checklist (and note appropriate exceptions) before the PR is marked ready-for-review.

- [ ] Changes are compatible[^1]
- [ ] Documentation[^2] completed
- [ ] Performance impact assessed and acceptable
- Tests added and passing[^3]
    - [ ] Unit Tests
    - [ ] Integration Tests
    - [ ] Manual Tests

**Exceptions**

*Note any exceptions here*

**Notes**

[^1]: It may be appropriate to bring upcoming changes to the attention of other (impacted) groups. Please endeavour to do this before seeking PR approval. The mechanism for doing this will vary considerably, so use your judgement as to how and when to do this.
[^2]: Configuration is an important part of many changes. Where applicable please try to document configuration examples.
[^3]: Tick whichever testing boxes are applicable. If you are adding Manual Tests, please document the manual testing (extensively) in the Exceptions.

[ROUTER-736]: https://apollographql.atlassian.net/browse/ROUTER-736?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: apollo-router/src/spec/query/transform.rs [hunk 1]
```diff
@@ -424,6 +417,35 @@ pub(crate) fn selection_set(
...
 }
 
+fn used_variables_from_value<V: Visitor>(
+    visitor: &mut V,
+    argument_value: &apollo_compiler::ast::Value,
+) {
+    println!(
+        "used_variables_from_value: argument value: {:?}",
+        argument_value
+    );
```
Comment (Geal) [inline apollo-router/src/spec/query/transform.rs:427]:
```suggestion
```

**Review by dariuszkuc — APPROVED** [review]

**Review by duckki — APPROVED** [review]
Looks good!

The other bug I worked on was related to operation-level directives and their arguments that reference variables. That case is handled by this PR.

BTW, I got curious if query variable definition's default values could refernce another query variable. But, that turned out to be prevented by the spec. So, we don't need to visit (operation's) variable definitions here.

File: apollo-router/src/spec/query/transform.rs [hunk 2]
```diff
@@ -827,5 +853,85 @@ fragment F on Query {
         let doc = ast::Document::parse(query, "query.graphql").unwrap();
         let result = document(&mut visitor, &doc).unwrap();
         insta::assert_snapshot!(TestResult { query, result });
+
+        // test removed field with variable in list argument
+        let query = r#"
+            query($a: String, $b: String) {
+                c
+                d(arg: ["a", $a, "b"]) @remove
```
Comment (lrlna) [inline apollo-router/src/spec/query/transform.rs:861]: i am going to be very pedantic 😅😅😅. 

can these tests also test for a variable in list in list, and a variable in object in a list. example (and not a semantically correct one, so just to illustrate a point):
* d(arg: ["a", [$a], "b"]) @remove
* d(arg: ["a", {c: $a}, "b"]) @remove

Comment (Geal) [reply apollo-router/src/spec/query/transform.rs:865]: sure thing: [ce12831](https://github.com/apollographql/router/pull/6025/commits/ce128316ff95394bb1a0017bd42b6a89990ee844)

**Comment by Geal** [comment]
(rebasing the PR to retargetet it to the 1.55.0 branch)

**Review by IvanGoncharov — APPROVED** [review]

**Pull request merged**